test/algorithms/classifiers/test_neural_network_classifier.py

```diff
@@ -104,25 +126,43 @@ def test_classifier_with_opflow_qnn(self, config):
         score = classifier.score(X, y)
         self.assertGreater(score, 0.5)
 
+        # callback
+        if callback is not None:
+            self.assertTrue(all(isinstance(value, float) for value in history["values"]))
+            for weights in history["weights"]:
+                self.assertTrue(all(isinstance(weight, float) for weight in weights))
```

Comment:
I'd add a check that the number of weighs is equal to `qnn.num_weights`.

Reply:
Added checks in https://github.com/Qiskit/qiskit-machine-learning/pull/151/commits/472ca8b0a20f9b860a098f6aeb8df7747bd2353e